src/Workleap.DomainEventPropagation.Publishing/EventPropagationClient.cs

```diff
@@ -33,25 +33,28 @@ public Task PublishDomainEventAsync(string subject, IDomainEvent domainEvent, Ca
         => this.PublishDomainEventAsync(typeof(T).FullName, domainEvent, cancellationToken);
 
     public Task PublishDomainEventsAsync<T>(IEnumerable<T> domainEvents, CancellationToken cancellationToken) where T : IDomainEvent
-        => this.PublishDomainEventsAsync(typeof(T).FullName, domainEvents as IEnumerable<IDomainEvent>, cancellationToken);
+        => this.PublishDomainEventsAsync(typeof(T).FullName, domainEvents, cancellationToken);
```

Comment:
This is interesting, having the subject set that way makes it that every event no matter their type will have the same subject. Which means we probably don't use the subject anywhere because given the doc I'd expect this field to have a unique value per event. It technically would be used to tell you where the additionnal data is 

https://learn.microsoft.com/en-us/azure/event-grid/event-schema#event-properties


Maybe it is beyond the scope of this story, but something we should revisit I feel